src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,867 @@
+        }
+
+        return numQueuedDeallocations;
+    }
+
+    /// @dev gets the latest total magnitude or overwrites it if it is not set
+    function _getLatestTotalMagnitude(address operator, IStrategy strategy) internal returns (uint64) {
+        (bool exists,, uint224 totalMagnitude) = _totalMagnitudeUpdate[operator][strategy].latestSnapshot();
+        if (!exists) {
+            totalMagnitude = WAD;
+            _totalMagnitudeUpdate[operator][strategy].push({key: uint32(block.timestamp), value: totalMagnitude});
+            operatorMagnitudeInfo[operator][strategy].freeMagnitude = WAD;
+        }
+
+        return uint64(totalMagnitude);
+    }
+
+    /// @dev gets the pending free magnitude available to add by completing numToComplete pending deallocations
+    /// and returns the next index to start from if completed.
+    function _getPendingFreeMagnitude(
+        address operator,
+        IStrategy strategy,
+        uint16 numToComplete,
+        uint192 nextIndex
+    ) internal view returns (uint64 freeMagnitudeToAdd, uint192 completed) {
+        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+        freeMagnitudeToAdd = 0;
+        while (nextIndex < pendingFreeMagnitudeLength && completed < numToComplete) {
+            PendingFreeMagnitude memory pendingFreeMagnitude = _pendingFreeMagnitude[operator][strategy][nextIndex];
+            // pendingFreeMagnitude is ordered by completableTimestamp. If we reach one that is not completable yet, then break
+            // loop until completableTimestamp is < block.timestamp
+            if (uint32(block.timestamp) < pendingFreeMagnitude.completableTimestamp) {
+                break;
+            }
+
+            // pending free magnitude can be added to freeMagnitude
+            freeMagnitudeToAdd += pendingFreeMagnitude.magnitudeDiff;
+            ++nextIndex;
+            ++completed;
+        }
+        return (freeMagnitudeToAdd, completed);
+    }
+
+    /**
+     * @notice Returns the allocation delay of an operator
+     * @param operator The operator to get the allocation delay for
+     */
+    function allocationDelay(
+        address operator
+    ) public view returns (bool isSet, uint32 delay) {
+        AllocationDelayInfo memory delayInfo = _allocationDelayInfo[operator];
+
+        bool pendingInEffect = delayInfo.pendingDelay != 0 && block.timestamp >= delayInfo.pendingDelayEffectTimestamp;
+
+        isSet = delayInfo.delay != 0 || pendingInEffect;
+
+        if (pendingInEffect) return (isSet, delayInfo.pendingDelay);
+
+        return (isSet, delayInfo.delay);
+    }
+
+    /**
+     * @param operator the operator to get the slashable magnitude for
+     * @param strategies the strategies to get the slashable magnitude for
+     *
+     * @return operatorSets the operator sets the operator is a member of and the current slashable magnitudes for each strategy
+     */
+    function getCurrentSlashableMagnitudes(
```

Comment:
this is basically a view function lib from here on.

@shrimalmadhur what methods are u using here, so we can move them to a periphery contract?